tests/testthat/test-workflow-bbi.R

```diff
@@ -305,5 +305,115 @@ withr::with_options(list(
+
+      # Passed config_path
+      nmtran_exe <- locate_nmtran(.config_path = file.path(MODEL_DIR_BBI, "bbi.yaml"))
+      # Confirm executable
+      expect_equal(as.character(nmtran_exe), "/opt/NONMEM/nm74gf/tr/NMTRAN.exe")
+      # Confirm NONMEM version
+      expect_equal(attr(nmtran_exe, "nonmem_version"), "nm74gf")
+
+      # Wrong nmtran_exe path passed
+      expect_error(
+        locate_nmtran(mod1, nmtran_exe = "/opt/NONMEM/nm74gf/tr/NMTRAN2.exe"),
+        "Could not find an NMTRAN executable"
+      )
+
+      # no configuration file found
+      expect_error(
+        locate_nmtran(.config_path = file.path(tempdir(), "bbi.yaml")),
+        "No bbi configuration was found"
+      )
+    })
+
+    it("execute_nmtran", {
+      # Execute in subdirectory to avoid messing with other tests
+      nmtran_dir <- file.path(MODEL_DIR_BBI, "nmtran")
+      fs::dir_create(nmtran_dir)
+      on.exit(fs::dir_delete(nmtran_dir), add = TRUE)
+
+      # Copy model file into new model dir
+      fs::file_copy(CTL_TEST_FILE, nmtran_dir, overwrite = TRUE)
+      mod1 <- new_model(file.path(nmtran_dir, "1"), .overwrite = TRUE)
+
+      # create new bbi.yaml
+      bbi_init(
+        nmtran_dir,
+        .nonmem_dir = Sys.getenv("BBR_TESTS_NONMEM_DIR", "/opt/NONMEM"),
+        .nonmem_version = Sys.getenv("BBR_TESTS_NONMEM_VERSION", "nm74gf"),
+        .bbi_args = list(mpi_exec_path = get_mpiexec_path())
+      )
+
+      nmtran_exe <- locate_nmtran(mod1)
+      nmtran_results <- execute_nmtran(
+        nmtran_exe, mod_path = basename(get_model_path(mod1)), dir = nmtran_dir
+      )
+
+      # Check attributes
+      expect_equal(nmtran_dir, nmtran_results$run_dir)
+      expect_equal(nmtran_results$status_val, 0)
+      expect_equal(nmtran_results$status, "NMTRAN successful")
+
+      # Test failure
+      data_path <- "test/this/path/data.csv"
+      modify_data_path_ctl(mod1, data_path)
+
+      nmtran_results <- execute_nmtran(
+        nmtran_exe, mod_path = basename(get_model_path(mod1)), dir = nmtran_dir
+      )
+
+      # Check attributes
+      expect_equal(nmtran_results$status_val, 4)
+      expect_equal(nmtran_results$status, "NMTRAN failed. See errors.")
+    })
+
+    it("run_nmtran: integration", {
+      # create model
+      mod1 <- read_model(file.path(MODEL_DIR_BBI, "1"))
+
+      nmtran_results <- run_nmtran(mod1, delete_on_exit = FALSE)
+      on.exit(fs::dir_delete(nmtran_results$run_dir))
+
+      # Check attributes
+      expect_equal(get_model_path(mod1), nmtran_results$absolute_model_path)
+      expect_equal(
+        file.path(nmtran_results$run_dir, basename(get_model_path(mod1))),
+        nmtran_results$nmtran_model
+      )
+      expect_equal(nmtran_results$nonmem_version, "nm74gf")
+      expect_equal(nmtran_results$status_val, 0)
+      expect_equal(nmtran_results$status, "NMTRAN successful")
+    })
+
+    it("nm_fdata: integration", {
+      fdata <- nm_fdata(mod1)
+
```

Comment:
```
$ Rscript -e 'devtools::test(filter = "workflow-bbi")'
ℹ Testing bbr
✔ | F W  S  OK | Context
✖ | 1       62 | testing a composable workflow and running bbi [101.3s]                                        
───────────────────────────────────────────────────────────────────────────────────────────────────────────────
Error ('test-workflow-bbi.R:406:7'): run_nmtran: nm_fdata: integration
Error in `eval(code, test_env)`: object 'mod1' not found
Backtrace:
    ▆
 1. └─bbr::nm_fdata(mod1) at test-workflow-bbi.R:406:7
 2.   └─bbr::run_nmtran(.mod, .config_path, nmtran_exe, delete_on_exit = FALSE) at bbr/R/run-nmtran.R:102:3
 3.     └─bbr:::check_model_object(.mod, "bbi_nonmem_model") at bbr/R/run-nmtran.R:48:3
 4.       └─base::inherits(.mod, .mod_types) at bbr/R/utils.R:407:3
───────────────────────────────────────────────────────────────────────────────────────────────────────────────
```

Reply:
Yeah this test was incomplete. Still have to add some other expectations to this test too